### DISCUSSION TITLE
FIDI V0.9.16 / Nordigen creditorName and ultimateCreditor Field

### DISCUSSION THREAD

**Discussion by matze338** [firefly-iii/firefly-iii#6469] [Firefly III]
I noticed during the last import that an account named "Adyen N.V." was created during the import from FIDI v0.9.16.
The Nordigen JSON string looks like this:

>            {
>                "transactionId": "xxxx",
>                "entryReference": "2022-09-20-08.52.13.824558",
>                "endToEndId": "xxxx",
>                "bookingDate": "2022-09-20",
>                "valueDate": "2022-09-20",
>                "transactionAmount": {
>                    "amount": "-76.97",
>                    "currency": "EUR"
>                },
>                "creditorName": "Adyen N.V.",
>                "creditorAccount": {
>                    "iban": "xxxx"
>                },
>                "ultimateCreditor": "DECATHLON Deutschland SE Co",
>                "remittanceInformationStructured": "2022-09-16T14:50      Debitk.4 2024-12",
>                "additionalInformation": "KARTENZAHLUNG",
>                "purposeCode": "IDCP",
>                "proprietaryBankTransactionCode": "xxxx",
>                "creditorAgent": "xxxx",
>                "debtorAgent": "xxxx"
>            },

The company I bought from is in the field "ultimateCreditor" and not "creditorName"!

In addition, I noticed that different companies probably make debits under the same IBAN and these also only differ by the "ultimateCreditor".

>            {
>                "transactionId": "xxxx",
>                "entryReference": "2022-09-20-09.17.17.963887",
>                "endToEndId": "xxxx",
>                "bookingDate": "2022-09-20",
>                "valueDate": "2022-09-20",
>                "transactionAmount": {
>                    "amount": "-26.98",
>                    "currency": "EUR"
>                },
>                "creditorName": "AWG",
>                "creditorAccount": {
>                    "iban": "DExxxxxxxxxxxx1052141"
>                },
>                "ultimateCreditor": "AWG Mode Center Fil.1337",
>                "remittanceInformationStructured": "2022-09-19T17:15      Debitk.4 2024-12",
>                "additionalInformation": "KARTENZAHLUNG",
>                "purposeCode": "IDCP",
>                "proprietaryBankTransactionCode": "xxxx",
>                "creditorAgent": "xxxx",
>                "debtorAgent": "xxxx"
>            },
>			
>            {
>                "transactionId": "xxxx",
>                "entryReference": "2022-09-28-09.12.10.151562",
>                "endToEndId": "xxxx",
>                "bookingDate": "2022-09-28",
>                "valueDate": "2022-09-28",
>                "transactionAmount": {
>                    "amount": "-11.75",
>                    "currency": "EUR"
>                },
>                "creditorName": "OSTERTAL-APOTHEKE",
>                "creditorAccount": {
>                    "iban": "DExxxxxxxxxxxx1052141"
>                },
>                "ultimateCreditor": "Ostertal-Apotheke",
>                "remittanceInformationStructured": "2022-09-26T17:32      Debitk.4 2024-12",
>                "additionalInformation": "KARTENZAHLUNG",
>                "purposeCode": "IDCP",
>                "proprietaryBankTransactionCode": "xxxx",
>                "creditorAgent": "xxxx",
>                "debtorAgent": "xxxx"
>            },
			
But such cases cannot be mapped correctly with Firefly?
Budget allocations do not work in such a case.

Looks like these are bookings that are processed via external service providers who use a single IBAN for their customers. In the case of Decathlon, for example, Adyen N.V. For my debit from "AWG" and "Ostertal-Apotheke" because of the same IBAN probably also.

Is the field "ultimateCreditor" saved during the import, so that a rule can be used to allocate the budget correctly?

Thanks

**Comment by JC5** [comment]
> But such cases cannot be mapped correctly with Firefly?

No, not yet. This is something I wish to add in the future. I'll make a ticket to track it!
